docker/docker_client.go

```diff
@@ -243,7 +243,13 @@ func (c *dockerClient) getBearerToken(realm, service, scope string) (string, err
 	return tokenStruct.Token, nil
 }
 
-func getAuth(hostname string) (string, string, error) {
+func getAuth(ctx *types.SystemContext, hostname string) (string, string, error) {
+	if ctx.DockerAuthConfig != nil {
+		if ctx.DockerAuthConfig.Auth != "" {
+			return decodeDockerAuth(ctx.DockerAuthConfig.Auth)
+		}
+		return ctx.DockerAuthConfig.Username, ctx.DockerAuthConfig.Password, nil
```

Comment:
(It is not _completely_ certain that we can always ignore the `hostname` here, but it is likely enough to be the common case; we can always add a `DockerAuthConfigMap` later, so this is fine.)